[Detection Engine] update API docs for new ES|QL rule (#4052)
elastic/security-docs: docs/detections/api/rules/rules-api-create.asciidoc

* *{ml-cap} rules*: Creates an alert when a {ml} job discovers an anomaly above
the defined threshold (see <<machine-learning>>).
* *New terms*: Generates an alert for each new term detected in source documents within a specified time range.

> * *ES|QL*: Uses the Elasticsearch Query Language (ES|QL) rule to find events and aggregate search results.

Author: agree on moving both new terms and ESQL up

Your suggestion uses `uses` twice in one sentence. Let's rephrase it

> * *ES|QL*: Uses the Elasticsearch Query Language (ES|QL) rule to find events and aggregate search results.

Reviewer: We can actually remove the first part that describes the rule type being used:

```suggestion
* *ES|QL*: Uses {ref}/esql.html[ES|QL] to find events and aggregate search results. 
```

> * *ES|QL*: Uses the Elasticsearch Query Language (ES|QL) rule to find events and aggregate search results.

Author: ```suggestion
* *ES|QL*: Uses {ref}/esql.html[Elasticsearch Query Language (ES|QL)] to find events and aggregate search results.
```

I think it would be beneficial, to decode ESQL, similarly like in EQL example

> NOTE: ES|QL rule is in technical preview stage

Author: looks like tag is not working

<img width="856" alt="Screenshot 2023-10-18 at 16 45 10" src="https://github.com/elastic/security-docs/assets/92328789/bff26e53-7ee4-4dbe-814c-97cd8cdd6d3e">

* <<req-fields-ml>>
* <<req-fields-threat-match>>
* <<req-fields-new-terms>>
* <<req-fields-esql>>
* <<opt-fields-all>>
* <<opt-fields-query-eql>>
* <<opt-fields-threat-match>>
* <<opt-fields-query-threshold>>
* <<opt-fields-eql-query-threshold>>
* <<opt-fields-eql-query-threshold-new_terms-esql>>

[[ref-fields-all]]

> ===== Required fields for all rule types

Reviewer: Oh yeah, scratch my comment about the index field. I'm so used to defining indices for the other rule types that I forget the data source is defined on the query level when creating/updating ES|QL rules. 

That said, do we still need to add the note about required metadata fields to [this spot](https://security-docs_4052.docs-preview.app.elstc.co/guide/en/security/master/rules-api-create.html#req-fields-query-threshold)? More specifically, do we need to tell users that they need to include the metadata operator if they're writing a non-aggregating ES|QL query?

> ===== Required fields for all rule types

Author: I don't think so. This should and will be covered in a separate section. Maybe give link to that section from here?
Otherwise, we will duplicate this info and it would be harder to maintain in future, if we change something

> ===== Required fields for all rule types

Reviewer: Yeah, we can add a line telling users to look at the [ES|QL rule docs](https://www.elastic.co/guide/en/security/master/rules-ui-create.html#create-esql-rule) for more info on writing ES|QL queries.

> ===== Required fields for all rule types

Reviewer: Resolved in [576067e](https://github.com/elastic/security-docs/pull/4052/commits/576067e0d7eee9eb46c4af9eb17e0c4e9e74f6f8).

|type |String a|Data type on which the rule is based:

* `eql`: EQL query (see {ref}/eql.html[Event Query Language]).

> * `esql`: ES|QL query (see {ref}/esql.html[Elasticsearch Query Language]).

Reviewer 2: ```suggestion
* `esql`: ES\|QL query (see {ref}/esql.html[Elasticsearch Query Language]).
```
In the HTML preview, this line renders as:
 * `esql`: ES

 I think we need to escape the pipe character since this content is inside a table.

|==============================================

[[req-fields-query-threshold]]
===== Required fields for query, indicator match, threshold, new terms, event correlation, and ES|QL rules

[width="100%",options="header"]
|==============================================
|Name |Type |Description

|query |String a|{kibana-ref}/search.html[Query] used by the rule to create
alerts. 

[NOTE]
======
* For indicator match rules, only the query's results are used to determine
whether an alert is generated. 

> * Refer to <<create-esql-rule,Create ES\|QL rules>> for query requirements. 

Reviewer: ```suggestion
* ES\|QL rules have additional query requirements. Refer to <<create-esql-rule,Create ES\|QL rules>> for more information. 
```

|==============================================

[[req-fields-query-threshold]]

> ===== Required field for query, indicator match, threshold, new terms, event correlation and, ES|QL rules

Reviewer 2: ```suggestion
===== Required field for query, indicator match, threshold, new terms, event correlation, and ES|QL rules
```

> ===== Required field for query, indicator match, threshold, new terms, event correlation, and ES|QL rules

Reviewer: ```suggestion
===== Required fields for query, indicator match, threshold, new terms, event correlation, and ES|QL rules
```

(*Kibana* → *Stack Management* → *Advanced Settings* →
`securitySolution:defaultIndex`).

|==============================================

[[opt-fields-eql-query-threshold-new_terms-esql]]

> ===== Optional fields for event correlation, query, threshold, new terms, and ES|QL rules

Reviewer: Can this section be combined with [the previous one](https://github.com/elastic/security-docs/pull/4052/files#diff-b06e37b58b218d307080f6dfb75f551ca3a09a5568a1b901f49947a7013e4e20R375)? I'm having trouble discerning which fields are applicable/optional for the event correlation, query, threshold, and new terms rules, and which are only applicable/optional for ES|QL rules.

Reviewer: Resolved in [5d69fe4](https://github.com/elastic/security-docs/pull/4052/commits/5d69fe43645fcac3a8527428d5392609b160b5bc).

`200`::
    Indicates a successful call.

*Example 7*

ES|QL rule that creates alerts from events that match an Excel parent process:

[source,json]
--------------------------------------------------
POST api/detection_engine/rules
{

>    "type":"esql",
>    "language":"esql",
>    "query":"from auditbeat-8.10.2 [metadata _id, _version, _index] | where process.parent.name == \"EXCEL.EXE\"",
>    "name":"Find Excel events",
>    "description":"Find Excel events",
>    "tags":[],
>    "interval":"5m",
>    "from":"now-360s",
>    "to":"now",
>    "enabled":false,
>    "risk_score": 21,
>    "severity": "low"

Reviewer 2: ```suggestion
  "type": "esql",
  "language": "esql",
  "query": "from auditbeat-8.10.2 [metadata _id, _version, _index] | where process.parent.name == \"EXCEL.EXE\"",
  "name": "Find Excel events",
  "description": "Find Excel events",
  "tags": [],
  "interval": "5m",
  "from": "now-360s",
  "to": "now",
  "enabled": false,
  "risk_score": 21,
  "severity": "low"
```

`200`::
    Indicates a successful call.

*Example 7*

> ES|QL rules that creates alerts from events that matches Excel parent process:

Reviewer 2: ```suggestion
ES|QL rule that creates alerts from events that match an Excel parent process:
```

[source,json]
--------------------------------------------------
POST api/detection_engine/rules
{
   "type":"esql",
   "language":"esql",

>    "query":"from auditbeat-8.10.2 [metadata _id] | where process.parent.name == \"EXCEL.EXE\"",

Reviewer: I thought all three metadata properties needed to be added to ensure alerts were de-duped, though maybe I'm misunderstanding that? Do users only need to specify the `_id` property to enable alert de-duping?

```suggestion
   "query":"from auditbeat-8.10.2 [metadata _id, _version, _index] | where process.parent.name == \"EXCEL.EXE\"",
```

}
--------------------------------------------------
<1> dev:[] These fields are under development and their usage may change: `related_integrations`, `required_fields`, and `setup`.

Example response for an ES|QL rule:

[source,json]
--------------------------------------------------
{

>     "name": "Find Excel events",
>     "description": "Find Excel events",
>     "risk_score": 21,
>     "severity": "low",
>     "output_index": "",
>     "tags": [],
>     "interval": "5m",
>     "enabled": false,
>     "author": [],
>     "false_positives": [],
>     "from": "now-360s",
>     "max_signals": 100,
>     "risk_score_mapping": [],
>     "severity_mapping": [],
>     "threat": [],
>     "to": "now",
>     "references": [],
>     "version": 1,
>     "exceptions_list": [],
>     "actions": [],
>     "id": "d0f20490-6da4-11ee-b85e-09e9b661f2e2",
>     "updated_at": "2023-10-18T10:55:14.269Z",
>     "updated_by": "elastic",
>     "created_at": "2023-10-18T10:55:14.269Z",
>     "created_by": "elastic",
>     "revision": 0,
>     "rule_id": "e4b53a89-debd-4a0d-a3e3-20606952e589",
>     "immutable": false,
>     "related_integrations": [],     <1>
>     "required_fields": [],          <1>
>     "setup": "",                    <1>
>     "type": "esql",
>     "language": "esql",
>     "query": "from auditbeat-8.10.2 [metadata _id] | where process.parent.name == \"EXCEL.EXE\""

Reviewer 2: ```suggestion
  "name": "Find Excel events",
  "description": "Find Excel events",
  "risk_score": 21,
  "severity": "low",
  "output_index": "",
  "tags": [],
  "interval": "5m",
  "enabled": false,
  "author": [],
  "false_positives": [],
  "from": "now-360s",
  "max_signals": 100,
  "risk_score_mapping": [],
  "severity_mapping": [],
  "threat": [],
  "to": "now",
  "references": [],
  "version": 1,
  "exceptions_list": [],
  "actions": [],
  "id": "d0f20490-6da4-11ee-b85e-09e9b661f2e2",
  "updated_at": "2023-10-18T10:55:14.269Z",
  "updated_by": "elastic",
  "created_at": "2023-10-18T10:55:14.269Z",
  "created_by": "elastic",
  "revision": 0,
  "rule_id": "e4b53a89-debd-4a0d-a3e3-20606952e589",
  "immutable": false,
  "related_integrations": [],     <1>
  "required_fields": [],          <1>
  "setup": "",                    <1>
  "type": "esql",
  "language": "esql",
  "query": "from auditbeat-8.10.2 [metadata _id] | where process.parent.name == \"EXCEL.EXE\""
```

}
--------------------------------------------------
<1> dev:[] These fields are under development and their usage may change: `related_integrations`, `required_fields`, and `setup`.

> Example response for a ES|QL rule:

Reviewer 2: ```suggestion
Example response for an ES|QL rule:
```
